src/schema/v2/me/deleteUserInterestMutation.ts

```diff
@@ -37,22 +44,32 @@ export const deleteUserInterestMutation = mutationWithClientMutationId<
       },
     },
   },
-  mutateAndGetPayload: async (args, { deleteUserInterestLoader }) => {
-    if (!deleteUserInterestLoader) {
+  mutateAndGetPayload: async (
+    args,
+    { meDeleteUserInterestLoader, deleteUserInterestLoader }
+  ) => {
+    if (!meDeleteUserInterestLoader || !deleteUserInterestLoader) {
       throw new Error("You need to be signed in to perform this action")
     }
 
+    // check if the delete operation is for another user
+    const isOnMe = args.isOnMe ?? true
+
     // snake_case keys for Gravity (keys are the same otherwise)
     const { id, ...gravityOptions } = Object.keys(args).reduce(
       (acc, key) => ({ ...acc, [snakeCase(key)]: args[key] }),
       {} as GravityInput
     )
 
     try {
-      const userInterest: UserInterest = await deleteUserInterestLoader?.(
-        id,
-        gravityOptions
-      )
+      let userInterest: UserInterest
+      if (isOnMe === false) {
+        // delete interest for another user
+        userInterest = await deleteUserInterestLoader?.(id, gravityOptions)
```

Comment:
Same point as above about breaking this out into its own schema. This would also remove the reliance on using a flag.